src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -163,12 +163,10 @@ private void addBinaryToPath(String hdfsBinaryPath) throws IOException {
    **/
   protected void startProcess(ExecutorDriver driver, Task task) {
     reloadConfig();
-    Process process = task.process;
-    if (process == null) {
+    if (task.process == null) {
       try {
-        process = Runtime.getRuntime().exec(new String[]{
-            "sh", "-c", task.cmd});
-        redirectProcess(process);
+        task.process = new ProcessBuilder("sh", "-c", task.cmd).start();
```

Comment:
Instead of method chaining, it is better java style to create a separate process builder object then call start on a separate line.